src/utils/AnalyticsUtils.ts

```diff
@@ -1,13 +1,64 @@
+import { history } from 'coveo.analytics';
 import { IAnalyticsActionCause, analyticsActionCauseList } from '../ui/Analytics/AnalyticsActionListMeta';
+import { findLastIndex } from 'underscore';
+
+export interface IClientInformationProvider {
+  readonly visitorId: string;
+  readonly clientId: string;
+  readonly disabled: boolean;
+}
 
 export class AnalyticsUtils {
-  static addActionCauseToList(newActionCause: IAnalyticsActionCause) {
+  private static currentInstance: AnalyticsUtils;
+  private clientInformationProvider: IClientInformationProvider = null;
+
+  public static get instance() {
+    return this.currentInstance || (this.currentInstance = new AnalyticsUtils());
+  }
```

Comment:
I'm not clear on why we need to use a `static` method to access a `static` property that is holding a instance of `AnalyticsUtils`. If someone wants AnalyticsUtils, could they not instantiate it directly?

Reply:
The goal was to have a singleton to restrict reading analytics if they weren't explicitly enabled by the instantiation of `Analytics`. Do you think it would be better to have a static reference to `Analytics` in `AnalyticsUtils` and make `AnalyticsUtils` a simple class?

Reply:
Hmm, since the cookie is effectively a singleton, I don't think we need to create another. If it's just to restrict access, I think a nicer approach is to either a) pass an option via the constructor when initializing, or b) create a Null version of the class following the same interface, that returns empty strings, and a function that is responsible for initializing the correct class. e.g.

```
interface IAnalyticsUtils {
    method1: ...
    method2: ...
}

function buildAnanlyticsUtils(options): IAnalyticsUtils {
    if (options.isAnanlyticsActive) {
        return new AnanlyticsUtils()
    }

    return new NullAnalyticsUtils()
}
```

Option `a` is more concise, whereas option `b` is a little cleaner since it extracts the conditional out of the class. I think option `a` is okay in this case, but I leave it to what you find best,

Reply:
After all, I went for the solution you suggested below (adding an `AnalyticsInformation` class).